src/components/Chart/Chart.js

```diff
@@ -0,0 +1,297 @@
+    }
+
+    handleHighlightEnter = (event, severity, key, index) => {
+        if (!this.state.rectIsHovered) {
+            console.log('handleHighlightEnter', this.state.rectIsHovered)
+            event.stopPropagation();
+            // console.log(severity, key, index)
+            const hoveredRect = {
+                'severity': severity,
+                'scenario': key,
+                'index': index
+            }
+            const { severities, quantileObj }  = this.state;
+            const { stat, statLabel, summaryStart, summaryEnd, width } = this.props;
+            // const formatDate = timeFormat('%b %d, %Y'); //timeFormat('%Y-%m-%d')
+            const median = quantileObj[stat][severity][key]['median']
+            const tenth = quantileObj[stat][severity][key]['tenth']
+            const ninetyith = quantileObj[stat][severity][key]['ninetyith']
+            // console.log(quantileObj[stat][severity][key])
+
+            const tooltipText = `<b>50%</b> chance of <b>${addCommas(Math.ceil(median))}</b> ${statLabel}<br><br> ` +
+                                // `from <b>${formatDate(summaryStart)}</b> to <b>${formatDate(summaryEnd)}</b> <br><br>` +
+                                `<b>90%</b> chance of <b>${addCommas(Math.ceil(tenth))} to ${addCommas(Math.ceil(ninetyith))}</b> ${statLabel} `
+                                // `from <b>${formatDate(summaryStart)}</b> to <b>${formatDate(summaryEnd)}</b>`
+            // console.log(tooltipText)
+            const tooltip = this.tooltipRef.current;
+            tooltip.innerHTML = tooltipText
+            // (i * (width / severities.length) - margin.left - margin.right) + this.state.xScale(key)
+            if (severity === 'high') {
+                tooltip.style.marginLeft = `${(1 * (width  - margin.left - margin.right) / severities.length) + this.state.xScale(key) - 300}px`
+            } else if (severity === 'med') {
+                tooltip.style.marginLeft = `${(2 * (width  - margin.left - margin.right) / severities.length) + this.state.xScale(key) - 300}px`
+            } else {
+                tooltip.style.marginLeft = `${(3 * (width  - margin.left - margin.right) / severities.length) + this.state.xScale(key) - 300}px`
+            }
+            this.setState({ hoveredRect, rectIsHovered: true })
+            this.props.handleCalloutInfo( statLabel, median, tenth, ninetyith, true );
+            this.props.handleScenarioHover( index );
+        }
+    }
+
+    handleHighlightLeave = () => {
+        const hoveredRect = {
+            'severity': '',
+            'scenario': '',
+            'index': 0
+        }
+        this.setState({ hoveredRect, rectIsHovered: false })
+        this.props.handleCalloutLeave();
+        this.props.handleScenarioHover( null );
+    }
+
+    drawSummaryStats = () => {
+        const barWidth = ((this.props.width / this.state.severities.length) / this.props.scenarios.length) - margin.left - margin.right;
```

Comment:
Looks like you're passing width down through every component starting from the MainContainer. That's a lot of passing. Instead of passing this width/height down, you can make the child components just fill the parent's width/height. You can do with with percentage based constraints or flexbox.